hive/glue-sync.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+source /usr/lib/hustler/bin/qubole-bash-lib.sh
+source /usr/lib/qubole/bootstrap-functions/common/utils.sh
+
+##
+# Installs Hive Glue Catalog Sync Agent
+# param1 - Region for AWS Athena. Defaults to us-east-1
+# Requires Hive 2.x
+#
+function install_glue_sync() {
+    aws_region=${1:-us-east-1}
+
+    is_master=$(nodeinfo is_master)
+    hive_version=$(nodeinfo hive_version)
+    if [[ ${is_master} == "1" && ${hive_version} == 2* ]]; then
+        glue_staging_dir=$(nodeinfo s3_default_location)
+        glue_staging_dir="${glue_staging_dir}/query_result"
+
+        hadoop dfs -get s3://paid-qubole/aws_glue_sync/HiveGlueCatalogSyncAgent-1.0-SNAPSHOT-jar-with-dependencies.jar /usr/lib/hive1.2/lib/
+
+        # Add glue sync configurations to hive-site.xml
+        # (Refer : https://github.com/awslabs/aws-glue-catalog-sync-agent-for-hive)
+        cp /usr/lib/hive1.2/conf/hive-site.xml /usr/lib/hive1.2/conf/hive-site.xml.bak
+        xmlstarlet ed --inplace --omit-decl -s '//configuration' -t elem -n "property" -v "" \
+            -s '//configuration/property[last()]' -t elem -n "name" -v "hive.metastore.event.listeners" \
+            -s '//configuration/property[last()]' -t elem -n "value" -v "com.amazonaws.services.glue.catalog.HiveGlueCatalogSyncAgent" /usr/lib/hive1.2/conf/hive-site.xml
+
+        xmlstarlet ed --inplace --omit-decl -s '//configuration' -t elem -n "property" -v "" \
+            -s '//configuration/property[last()]' -t elem -n "name" -v "glue.catalog.athena.jdbc.url" \
+            -s '//configuration/property[last()]' -t elem -n "value" -v "jdbc:awsathena://athena.${aws_region}.amazonaws.com:443" /usr/lib/hive1.2/conf/hive-site.xml
+
+        xmlstarlet ed --inplace --omit-decl -s '//configuration' -t elem -n "property" -v "" \
+            -s '//configuration/property[last()]' -t elem -n "name" -v "glue.catalog.athena.s3.staging.dir" \
+            -s '//configuration/property[last()]' -t elem -n "value" -v "${glue_staging_dir}" /usr/lib/hive1.2/conf/hive-site.xml
+
+        # Restart metastore
+        monit unmonitor metastore1_2
+        export OVERRIDE_HADOOP_JAVA_HOME=/usr/lib/jvm/java-1.8.0_60
```

Comment:
I think we should use java-1.8.0 and here and not specify the micro version. That is subject to change.

Reply:
Done